### PR TITLE
Revenant Rework [WIP | DNM | TEMPORARILY CLOSED]

### PR DESCRIPTION
**Pull request by ghost**
This rework will make revenants more focused on utilizing the environment around them. For instance, instead of dealing straight brute damage, they may cause electronic malfunctions or mental instability in the target.

This is largely work-in-progress.

### CODE REVIEW

File: code/modules/mob/living/simple_animal/revenant/revenant_abilities.dm [hunk 1]
```diff
@@ -142,49 +141,48 @@
...
+					L.visible_message("<span class='warning'>\The [L]'s light suddenly becomes a deep violet...</span>")
 					sleep(20)
 					for(var/mob/living/M in orange(4, L))
 						if(M == user)
 							return
-						M.Beam(L,icon_state="lightning",icon='icons/effects/effects.dmi',time=5)
-						M.electrocute_act(25, "[L.name]")
+						M.Stun(1)
+						M.Beam(L,icon_state="lightning",icon='icons/effects/effects.dmi',time=2)
+						M.electrocute_act(rand(5,30), "[L.name]")
```
Comment (Anonus) [inline code/modules/mob/living/simple_animal/revenant/revenant_abilities.dm]: I wish we had an actual lightning proc.

**Comment by AlexCrimson** [comment]
Really would be nice if you could add better feedback on damage the rev receives. You get no indication of how many "lives" you have. AoE attacks also seem to hit the rev when its invisible. I got killed by a repulse spell on the escape shuttle.

A simple UI that shows how much essence you have would be great. 

Ability wise the break windows seems fun. Some kind of ability that affects the targets blood system maybe, as i suggested a while ago. Lowering the temperature of an area would be neat n spooky. Also make Transmit free as its kinda useless for anything other than RP. Oh, and force choke pls.

One thing im worried about is essence. It was hard enough to gather it before. But now you will basically only have indirect methods to hurt someone, i feel like most Revenants will be sitting at the 25 essence point and using cheap abilities 90% of the time.

**Comment by Supermichael777** [comment]
well the stupid mind spike could kill basically anyone with no chance of retaliation.  as bad as ninja invis

**Comment by AlexCrimson** [comment]
Some more random dumb ideas...

Bloodlust - Gives the target temporary antag status to do as much damage to humanoids as possible for 60 seconds. Cannot be used on people with loyalty implants. If you do not hurt people then maybe you take damage or blood loss? That probably isnt possible code-wise.

Rather than allowing the Revenant to wonder around anywhere on the map, have it "haunt" areas. It can only move within the area it haunts. Haunting a new area is free but takes a few seconds to teleport. The Chaplin would maybe get a message when he enters an area that is haunted. Maybe have a really expensive ability that allows it to haunt the entire station.

An ability that sprays blood in an area. Would be spooky. Maybe make standing in the blood do damage or inject some kind of hallucinogen/blindness chemical.

There needs to be some actual counters to Revenant before its re-added. The holy water/salt thing needs to be implemented in some way. Just hitting the thing is too hard. Unless a Revenant starts harvesting someone in plain sight.

Some kind of targeted stun ability so the Revenant can interfere and influence murders. A Sec officer about to stun a tator? Stun the officer and let the tator kill him!

**Comment by Xhuis** [comment]
Closing because I have a lot more to do. Will re-open later.

**Comment by AlexCrimson** [comment]
RIP

**Comment by palpatine213** [comment]
Not sure why, but it seems mind spike alone isn't enough to crit someone. It seems to respect the damage cap on limbs. I kept hitting one guy over and over, and the only thing that got him to crit was the oxy damage taken from bleeding out.

**Pull request closed without merging**